tests/span_tests.cpp

```diff
@@ -54,7 +54,6 @@ struct DerivedClass : BaseClass
 };
 struct AddressOverloaded
 {
-    AddressOverloaded operator&() const { return {}; }
```

Comment:
This `operator&` overload is the *raison dêtre* for having this type and using it in the tests - it's to ensure that everything which should be using `std::addressof` instead of `&` is doing so.

Reply:
You *could* add a call somewhere specifically to silence the warning and avoid all the boilerplate, but I agree that the boilerplate makes it blatantly clear that this function is unused for good reason.

Reply:
This being the test suite I think the boilerplate would be acceptable.